### DISCUSSION TITLE
PKCS#11 Option is not available in ejbca community

### DISCUSSION THREAD

**Discussion by sysadminbioenable** [Keyfactor/ejbca-ce#671] [General]
I’m currently working with the EJBCA Community Edition and I’ve encountered an issue regarding the creation of Crypto Tokens. Specifically, when I navigate to Crypto Tokens > Create New, I only see the option for Soft tokens and not the PKCS#11 option.
From my understanding, the PKCS#11 option should be available if EJBCA can detect any known PKCS#11 modules in the file system. However, despite checking the documentation and ensuring that the necessary modules are installed, the option still doesn’t appear.

Note: I have deployed  EJBCA Community Edition using Docker (keyfactor/ejbca-ce) image.

**Comment by primetomas** [comment]
You need P11 drivers available in order for PKCS#11 option to show up.
https://docs.keyfactor.com/ejbca/latest/hardware-security-modules-hsm
https://github.com/Keyfactor/keyfactorcommunity
